### DISCUSSION TITLE
The first create_engine operation takes longer time in several create_engine operations in a row

### DISCUSSION THREAD

**Discussion by huanyu-li** [sqlalchemy/sqlalchemy#6667] [Usage Questions]
Hi,

I'm new to sqlalchemy and may I ask you a question regarding the create_engine() function? 
I have to call create_engine multiple times based on different schema.
I just observe that the first call takes a longed time. Is it reasonable and unavoidable?
(I'm using mysql as underlying database and the sqlalchemy version is 1.4.19.)

Thanks,
Best,
Huanyu

**Comment by zzzeek** [comment]
there's a connection pool in use, so if the first connect takes a long time you may have a network /database issue that is causing initial connection to take a long time.  there are also some queries that the engine runs up front when it first connects, these are very small and fast queries but if you have an extremely slow network connection or your database is having problems responding, this could potentially take a long time.

all indications point to checking out your network connectivity and database responsiveness.

**Reply by huanyu-li** [reply]
Thank you so much. I get the point.
